### PR TITLE
Port BTC single sig address api call to Rust

### PR DESCRIPTION
**Pull request by benma**

### CODE REVIEW

**Review by x1ddos — APPROVED** [review]
LGTM

File: src/rust/bitbox02-rust/src/hww/api.rs [hunk 1]
```diff
@@ -111,6 +127,7 @@ async fn process_api(request: &Request) -> Option<Result<Response, Error>> {
         #[cfg(not(feature = "app-ethereum"))]
         Request::Eth(_) => Some(Err(Error::Disabled)),
 
+        request @ Request::BtcPub(_) => process_api_btc(request).await,
```
[the remark this answers is not in the document]
Comment (benma) [reply src/rust/bitbox02-rust/src/hww/api.rs:130]: BtcPub and all btc signing messages precede the introduction of BTCRequest :(

File: src/rust/bitbox02-rust/src/hww/api.rs [hunk 2]
```diff
@@ -80,6 +83,19 @@ fn request_tag(request: &Request) -> u32 {
...
+async fn process_api_btc(request: &Request) -> Option<Result<Response, Error>> {
+    match request {
+        Request::BtcPub(ref request) => bitcoin::process_pub(request).await,
+        _ => None,
+    }
+}
+
+#[cfg(not(any(feature = "app-bitcoin", feature = "app-litecoin")))]
+async fn process_api_btc(_request: &Request) -> Option<Result<Response, Error>> {
+    Some(Err(Error::Disabled))
```
[the remark this answers is not in the document]
Comment (benma) [reply src/rust/bitbox02-rust/src/hww/api.rs:96]: The difference is due to the fact that all ETH requests are inside `ETHRequest`, while in BTC, some are toplevel (`BTCPub, `BTCSignInit`, ...), and some are in `BTCRequest`, so unfortunately it cannot be totally consistent.

File: src/rust/bitbox02-rust-c/Cargo.toml [hunk 3]
```diff
@@ -44,8 +44,8 @@ target-bootloader-btc-production = ["bootloader", "platform-bitbox02"]
 target-bootloader-bitboxbase = ["bootloader", "platform-bitboxbase"]
 target-bootloader-bitboxbase-development = ["bootloader", "platform-bitboxbase"]
 target-bootloader-bitboxbase-production = ["bootloader", "platform-bitboxbase"]
-target-firmware = ["firmware", "platform-bitbox02", "app-ethereum"]
-target-firmware-btc = ["firmware", "platform-bitbox02"]
+target-firmware = ["firmware", "platform-bitbox02", "app-bitcoin", "app-litecoin", "app-ethereum"]
+target-firmware-btc = ["firmware", "platform-bitbox02", "app-bitcoin"]
```
[the remark this answers is not in the document]
Comment (benma) [reply src/rust/bitbox02-rust-c/Cargo.toml:48]: Hm good point, not sure what's up with that. Removed and verified that the unit tests run for all features.

**Comment by benma** [comment]
> toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit

ouch.

File: src/rust/bitbox02-rust-c/Cargo.toml (hunk 3 above)
[the remark this answers is not in the document]
Comment (benma) [reply src/rust/bitbox02-rust-c/Cargo.toml:48]: Actually I was wrong, it was used in the C unit tests, as the name suggests :see_no_evil: restoring and adding the other apps...

**Pull request merged**